src/main/kotlin/com/atlassian/performance/tools/awsinfrastructure/api/network/access/LocalPublicIpv4Provider.kt

```diff
@@ -0,0 +1,61 @@
+package com.atlassian.performance.tools.awsinfrastructure.api.network.access
+
+import com.atlassian.performance.tools.awsinfrastructure.Ipv4Validator
+import com.atlassian.performance.tools.awsinfrastructure.VerboseUrlReader
+import java.net.URL
+import java.util.function.Supplier
+
+class LocalPublicIpv4Provider private constructor(
```

Comment:
What's local about it? Doesn't it provide IP as seen by the `servicesToQuery`?

Reply:
The "Local" part of name of this class was to indicate what instance this public IP is resolved for. In theory there could be a public IP address resolver for remote instances.

Do you have any other name suggestions for this class?

Side topic: I was thinking this weekend about class naming and came into a conclusion that implementations shouldn't have names. What we are naming is the contract which should be represented by an interface. IMHO it would be better to name implementations as `<interface name>V1`,`<inteface name>V2` and so on. I don't think we should start naming classes like that right now, but maybe this is something to brainstorm and consider for other projects.

Reply:
Aaah, I thought it's a "local provider of public ipv4", but it's a "provider of local public ipv4". All good :+1:

Reply:
> implementations shouldn't have names

Obviously current interfaces wouldn't fit: `SupplierOfStringV1`, `SupplierOfStringV2`, `SupplierOfStringV1329`. We'd have to capture all behaviors and aspects in the interface itself (e.g. local instance + public ipv4 + services to query + ip validation). I met an argument that variable names shouldn't be necessary, because the type should be specific enough to have only one possible meaning. These are interesting ideas to explore and experiment with, but I wouldn't bake them into API just yet.

Reply:
> Obviously current interfaces wouldn't fit

Yup, I didn't create a dedicated interface for it, because of laziness. Also it felt unnecessary.